ui/src/nav/MobileGroupRoot.tsx

```diff
@@ -1,27 +1,46 @@
-import CaretLeft16Icon from '@/components/icons/CaretLeft16Icon';
-import CaretRight16Icon from '@/components/icons/CaretRight16Icon';
-import ChannelList from '@/groups/GroupSidebar/ChannelList';
-import { useGroupFlag } from '@/state/groups';
 import React from 'react';
-import { Link } from 'react-router-dom';
+import { Link, useLocation } from 'react-router-dom';
+import { useGroupFlag, useGroup, useAmAdmin } from '@/state/groups';
+import ChannelList, { ChannelSorter } from '@/groups/GroupSidebar/ChannelList';
+import GroupAvatar from '@/groups/GroupAvatar';
+import CaretLeft16Icon from '@/components/icons/CaretLeft16Icon';
+import AddIcon from '@/components/icons/AddIcon';
 
 export default function MobileGroupRoot() {
   const flag = useGroupFlag();
+  const group = useGroup(flag);
+  const isAdmin = useAmAdmin(flag);
+  const location = useLocation();
 
   return (
     <>
-      <header className="flex flex-none items-center   justify-between px-2 py-1">
-        <Link
-          to="/"
-          className="default-focus inline-flex items-center rounded-lg p-2 text-base font-semibold text-gray-800 hover:bg-gray-50"
-        >
-          <CaretLeft16Icon className="mr-1 h-4 w-4 text-gray-400" />
-          Channels
-        </Link>
-        <Link to="./channels" className="small-secondary-button pr-1">
-          <span>All</span>
-          <CaretRight16Icon className="ml-1 h-4 w-4 text-gray-400" />
-        </Link>
+      <header className="flex items-center justify-between p-4 pt-3 pr-5">
+        <div>
+          <Link
+            to="/"
+            className="default-focus inline-flex items-center text-base font-semibold text-gray-800 hover:bg-gray-50"
+          >
+            <CaretLeft16Icon className="mr-2 h-6 w-6 text-gray-400" />
```

Comment:
use `CaretLeftIcon` if you're going to size it to 24px, convention so far has been only ever size down otherwise it'll look too thick as it does like this.